lib/FrontendTool/FrontendTool.cpp

```diff
@@ -2178,6 +2175,18 @@ int swift::performFrontend(ArrayRef<const char *> Args,
     return finishDiagProcessing(1);
   }
 
+  // The verifier must be setup after CompilerInstance::setup is called above to
+  // ensure the input buffer IDs have been added.
+  std::unique_ptr<DiagnosticVerifier> verifier;
+  if (diagOpts.VerifyMode != DiagnosticOptions::NoVerify) {
```

Comment:
This seems like a really tricky initialization condition - maybe we should move the verifier instance onto CompilerInstance and set it up in the right place then.

Reply:
Good idea, moving it onto CompilerInstance seems to work well